lib/generators/hyrax/work/work_generator.rb

```diff
@@ -1,6 +1,9 @@
 require 'rails/generators'
 
 class Hyrax::WorkGenerator < Rails::Generators::NamedBase
+  PLURAL_MODEL_NAME_WARN_MESSAGE = "[EXITING] The work type name '%s' was recognized as a plural, using the singular '%s' instead. " \
```

Comment:
I don't think so, since the plural-recognition is language-based.  To me, we should just be avoiding building anything that attempts to solve this problem.  It is not tractable and it is not our domain space.